common/test/acceptance/tests/studio/test_studio_html_editor.py

```diff
@@ -100,3 +113,186 @@ def test_link_plugin_sets_url_correctly(self):
             static_link,
             "URL in the link plugin is different"
         )
+
+    def test_tinymce_and_codemirror_preserve_style_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve style tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<p class='title'>pages</p><style><!-- .title { color: red; } --></style>" in the code editor and
+            press OK
+            And I save the page
+                Then the page text contains:
+                  ""
+                  <p class="title">pages</p>
+                  <style><!--
+                  .title { color: red; }
+                  --></style>
+                  ""
+        """
+        content = '<p class="title">pages</p><style><!-- .title { color: red; } --></style>'
+        self._content_test(content)
+
+    def test_tinymce_and_codemirror_preserve_span_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve span tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<span>Test</span>" in the code editor and press OK
+            And I save the page
+                Then the page text contains:
+                ""
+                    <span>Test</span>
+                ""
+        """
+        content = "<span>Test</span>"
+        self._content_test(content)
+
+    def test_tinymce_and_codemirror_preserve_math_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve math tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<math><msup><mi>x</mi><mn>2</mn></msup></math>" in the code editor and press OK
+            And I save the page
+                Then the page text contains:
+                ""
+                    <math><msup><mi>x</mi><mn>2</mn></msup></math>
+                ""
+        """
+        content = "<math><msup><mi>x</mi><mn>2</mn></msup></math>"
+        self._content_test(content)
+
+    def test_code_format_toolbar_wraps_text_with_code_tags(self):
+        """
+        Scenario: Code format toolbar button wraps text with code tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And I set the text to "display as code" and I select the text
+            And I save the page
+                Then the page text contains:
+                ""
+                    <p><code>display as code</code></p>
+                ""
+        """
+        self.html_editor.set_text_and_select("display as code")
+        self.html_editor.click_code_toolbar_button()
+        self.html_editor.save_content()
+        html = self.container_page.content_html
+        self.assertIn('<p><code>display as code</code></p>', html)
```

Comment:
the assertion would be more meaningful if we check presence of html in the result of our function instead of other way around
```suggestion
        self.assertIn('<p><code>display as code</code></p>', html)
```